src/vkscript/command_parser.cc

```diff
@@ -686,6 +705,22 @@ Result CommandParser::ProcessUniform() {
       cmd->SetBinding(val);
     }
 
+    {
+      // Generate an internal buffer for this binding if needed.
+      auto set = cmd->GetDescriptorSet();
+      auto binding = cmd->GetBinding();
+
+      auto* buffer = pipeline_->GetBufferForBinding(set, binding);
+      if (!buffer) {
+        auto b = MakeUnique<Buffer>(BufferType::kStorage);
```

Comment:
Change `BufferType::kStorage` to `BufferType::kUniform`

Reply:
Good catch, thanks.